src/control/lib/hostlist/hostset.go

```diff
@@ -94,15 +113,38 @@ func (hs *HostSet) Delete(stringHosts string) (int, error) {
 	return int(hs.list.hostCount - startCount), nil
 }
 
+// MergeSet merges the supplied HostSet into this one.
+func (hs *HostSet) MergeSet(other *HostSet) error {
+	hs.initList()
```

Comment:
Seems like hs.initList() should be called after the check for the error case.  Is there a good reason to call it before?

Reply:
Dunno if it's a "good" reason, but it's consistent with the other methods. It's true that there's potentially a slight temporary waste of cpu/memory in the case that other is nil, but the upside is that there is a pattern established of always ensuring that the hostlist is initialized at the top of every method. If this is forgotten, a nil pointer dereference crash may occur.

Reply:
Ok, it's certainly not harmful to initialize it here even if it's not used.  Sometimes our group gets concerned with these minor details and sometimes not.  I am happy to leave it as-is.